web/server/watcher.py

```diff
@@ -0,0 +1,30 @@
+import asyncio
+import json
+from pathlib import Path
+
+from watchfiles import awatch
+
+from sqlmesh.core import constants as c
+from web.server.api.endpoints.models import get_models
+from web.server.settings import get_loaded_context, get_settings
+from web.server.sse import Event
+
+
+async def watch_project(queue: asyncio.Queue) -> None:
+    settings = get_settings()
+    context = await get_loaded_context(settings)
+    ignore_patterns = context.config.ignore_patterns if context else c.IGNORE_PATTERNS
+
+    async for changes in awatch((context.path / "models").resolve()):
+        context.load()
+
+        output_models = []
+
+        for _, path in changes:
+            if any(Path(path).match(pattern) for pattern in ignore_patterns):
```

Comment:
I haven't looked too much into it yet, but is it possible to use watchfiles's [filters](https://watchfiles.helpmanual.io/api/filters/) instead?